Classes/Event/IndexQueue/AfterIndexQueueItemHasBeenMarkedForReindexingEvent.php

```diff
@@ -27,7 +27,7 @@ final class AfterIndexQueueItemHasBeenMarkedForReindexingEvent
 {
     public function __construct(
         private readonly string $itemType,
-        private readonly int|string $itemUid,
+        private readonly int $itemUid,
```

Comment:
To be more flexible and independent from the type of items to index a non-numeric item id must be supported.

This is also a preparation for indexing external data, the QueueInterface explicitly supports strings and integer values as item ids, if this is an issue in other Solr extensions we might have to reconsider the concept.

Reply:
OK,
i'll not merge this PR.